src/validation.cpp

```diff
@@ -1278,11 +1278,9 @@ void UpdateCoins(const CTransaction& tx, CCoinsViewCache& inputs, CTxUndo &txund
             // mark an outpoint spent, and construct undo information
             txundo.vprevout.push_back(CTxInUndo(coins->vout[nPos]));
             coins->Spend(nPos);
-            if (coins->vout.size() == 0) {
-                CTxInUndo& undo = txundo.vprevout.back();
-                undo.nHeight = coins->nHeight;
-                undo.fCoinBase = coins->fCoinBase;
-            }
+            CTxInUndo& undo = txundo.vprevout.back();
+            undo.nHeight = coins->nHeight;
+            undo.fCoinBase = coins->fCoinBase;
```

Comment:
In commit "Store/allow tx metadata in all undo records"

> Older code requires that there are 0s for everything but the last spend of an output, in the disconnect code. That's what the "allow" refers to in the commit title: it allows nonzero metadata in any undo record.

Could you clarify in the commit message what "won't be readable by older versions anymore" means? Will older software just log errors, or require re-indexing, or will it not work at all?

I understand that older code has a check for this condition here:
 https://github.com/bitcoin/bitcoin/blob/431a548faaf51c7a5fc89b6e479187a1c0e29805/src/validation.cpp#L1508. But I also thought this was one of the cases that DisconnectBlock would "graciously deal with" (according to earlier commit message).